### PR TITLE
Feature/qbittorrent api v2

### PR DESCRIPTION
**Pull request by p0psicles**
- [x] PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [x] Read the [contribution guide](https://github.com/pymedusa/Medusa/blob/master/.github/CONTRIBUTING.md)

Implement #6505

### CODE REVIEW

**Comment by p0psicles** [comment]
@sharkykh i tested this with a recent qbittorrent docker conainer. Seems to work fine. Anything I should be aware of?

**Comment by sharkykh** [comment]
Sorry, I don't remember what didn't work for me 15 months ago...

File: medusa/clients/torrent/qbittorrent.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 """qBittorrent Client."""
 
-from __future__ import unicode_literals
+from __future__ import division, unicode_literals
```
Comment (sharkykh) [inline medusa/clients/torrent/qbittorrent.py]: I was wrong, `division` is not needed in this case, as it's only for the normal division (`/`) operator, and does not affect modulo (`%`).

File: medusa/clients/torrent/qbittorrent.py [hunk 2]
```diff
@@ -36,30 +36,73 @@ def __init__(self, host=None, username=None, password=None):
...
         """Get API version."""
+        # Update the auth method to v2
+        self._get_auth = self._get_auth_v2
+        # Attempt to get API v2 version first
+        self.url = '{host}api/v2/app/webapiVersion'.format(host=self.host)
+        try:
+            version = self.session.get(self.url, verify=app.TORRENT_VERIFY_CERT,
+                                       cookies=self.session.cookies)
+            # Make sure version is using the (major, minor, release) format
+            version = map(int, version.content.split('.'))
```
Comment (sharkykh) [inline medusa/clients/torrent/qbittorrent.py]: I should have:
`from six.moves import map`
and
```suggestion
            version = list(map(int, version.text.split('.')))
```
----
or
```suggestion
            version = [int(v) for v in version.text.split('.')]
```

File: medusa/clients/torrent/qbittorrent.py [hunk 3]
```diff
@@ -36,30 +36,73 @@ def __init__(self, host=None, username=None, password=None):
...
+            return None
 
-        else:
+        if self.response.status_code == 404:
+            return None
+
+        self.session.cookies = self.response.cookies
+        self.auth = self.response.content
+
+        return self.auth if not self.response.status_code == 404 else None
```
Comment (sharkykh) [inline medusa/clients/torrent/qbittorrent.py]: The check is not needed
```suggestion
        return self.auth
```

**Comment by h2xtreme** [comment]
FYI I tested the latest qbittorrent associated with this issue and it works as expected.

`2019-08-11 02:01:38 WARNING	TORNADO :: [74c3f12] Your client doesn't support moving torrents to new location
2019-08-11 02:01:38 INFO	TORNADO :: [74c3f12] Successfully processed.
2019-08-11 02:01:38 INFO	TORNADO :: [74c3f12] Processing succeeded for /media/plex/downloads/medusa_postpro/WWE.205.Live.2019.08.06.1080p.WEB.h264-HEEL.mp4
2019-08-11 02:01:36 INFO	TORNADO :: [74c3f12] Selected poster for series 323420 (score=4.25, score_bay=7.5, rating=10, votes=1, res=680x1000): https://www.thetvdb.com/banners/posters/323420-1.jpg
2019-08-11 02:01:36 INFO	TORNADO :: [74c3f12] Selected fanart for series 323420 (score=4.58523929560476, score_bay=7.17047859120952, rating=10, votes=2, res=1920x1080): https://www.thetvdb.com/banners/fanart/original/323420-3.jpg
2019-08-11 02:01:29 INFO	TORNADO :: [74c3f12] Trying to clean any empty folder under /media/plex/Wrestling/WWE 205 Live
2019-08-11 02:01:29 INFO	TORNADO :: [74c3f12] This download is marked a priority download so I'm going to replace an existing file if I find one
2019-08-11 02:01:29 INFO	TORNADO :: [74c3f12] New file is the same size.
2019-08-11 02:01:29 INFO	TORNADO :: [74c3f12] New size: 1.91 GB
2019-08-11 02:01:29 INFO	TORNADO :: [74c3f12] Old size: 1.91 GB
2019-08-11 02:01:29 INFO	TORNADO :: [74c3f12] New file: /media/plex/downloads/medusa_postpro/WWE.205.Live.2019.08.06.1080p.WEB.h264-HEEL.mp4
2019-08-11 02:01:29 INFO	TORNADO :: [74c3f12] Old file: /media/plex/Wrestling/WWE 205 Live/WWE.205.Live.2019-08-06.August.6.2019.1080p_WEB.HEEL.mp4
2019-08-11 02:01:28 INFO	TORNADO :: [74c3f12] Processing /media/plex/downloads/medusa_postpro/WWE.205.Live.2019.08.06.1080p.WEB.h264-HEEL.mp4
2019-08-11 02:00:09 INFO	SHOWUPDATER :: [74c3f12] Completed but there was nothing to update
2019-08-11 02:00:00 INFO	SHOWUPDATER :: [74c3f12] Started periodic show updates
2019-08-11 02:00:00 INFO	SHOWUPDATER :: [74c3f12] Finished processing 3097 scene exceptions.
2019-08-11 02:00:00 INFO	SHOWUPDATER :: [74c3f12] Updating exception_cache and exception_season_cache
2019-08-11 01:58:09 INFO	SNATCHQUEUE-SNATCH-323420 :: [74c3f12] Downloading WWE.205.Live.2019.08.06.1080p.WEB.h264-HEEL with 1 seeders and 1 leechers and size 1.91 GB from NZBHydra2 - Torrents, through a SearchType.MANUAL_SEARCH search
2019-08-11 01:58:09 INFO	SNATCHQUEUE-SNATCH-323420 :: [74c3f12] Beginning to snatch release: WWE.205.Live.2019.08.06.1080p.WEB.h264-HEEL`

The only warning that shows up is a notification that qbitorrent doesn't support move completed downloads to another directory which given the way that qbittorrent works, IMO, is perfectly fine (this should be done either via custom script or by changing the torrent label after download is complete like Sonarr does).

Anyway the meat and potatoes works like a charm. thanks!

**Comment by p0psicles** [comment]
@medariox?

**Comment by medariox** [comment]
Will try to review tomorrow.

**Pull request merged**